analysis/Analysis/Section_3_4.lean

```diff
@@ -212,7 +230,22 @@ theorem SetTheory.Set.powerset_of_triple (a b c x:Object) :
     ∨ x = ({a,b}:Set)
     ∨ x = ({a,c}:Set)
     ∨ x = ({b,c}:Set)
-    ∨ x = ({a,b,c}:Set) := by sorry
+    ∨ x = ({a,b,c}:Set) := by
+  simp only [mem_powerset, subset_def, mem_triple]
+  constructor
+  · rintro ⟨Y, rfl, hY⟩
+    by_cases ha : a ∈ Y <;>
+    by_cases hb : b ∈ Y <;>
+    by_cases hc : c ∈ Y
+    · right; right; right; right; right; right; right; congr; apply Set.ext; simp; grind
```

Comment:
I've tried a few different things, and `grind` is the only concise *and* fast approach I could find so far. The other approach that works is `aesop` (with a helper lemma about `x \in Y \and y \nin Y \imp x \neq Y`), but `aesop` is *extremely* slow here and blows past the heartbeat limit.

I figured `grind` is fine despite the warning because it's [encouraged for new developments](https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/.60grind.60.20is.20available.2E.20What.20should.20Mathlib.20users.20expect.3F/near/526510167) starting with `v4.22.0-rc2`. We haven't updated to that yet but I presume we can soon.